v2/processor.go

```diff
@@ -151,41 +157,60 @@ type peekLockRenewer struct {
 }
 
 func (plr *peekLockRenewer) startPeriodicRenewal(ctx context.Context, message *azservicebus.ReceivedMessage) {
-	// _, span := tracing.StartSpanFromMessageAndContext(ctx, "go-shuttle.peeklock.startPeriodicRenewal", message)
-	// defer span.End()
 	count := 0
 	for alive := true; alive; {
 		select {
 		case <-time.After(*plr.renewalInterval):
-			log("renewing lock")
+			log(ctx, "renewing lock")
 			count++
-			// tab.For(ctx).Debug("Renewing message lock", tab.Int64Attribute("count", int64(count)))
+			tab.For(ctx).Debug("Renewing message lock", tab.Int64Attribute("count", int64(count)))
 			err := plr.lockRenewer.RenewMessageLock(ctx, message, nil)
 			if err != nil {
-				log("ERROR failed to renew lock")
-				// listener.Metrics.IncMessageLockRenewedFailure(message)
+				log(ctx, "failed to renew lock: ", err)
+				metrics.Processor.IncMessageLockRenewedFailure(message)
 				// I don't think this is a problem. the context is canceled when the message processing is over.
 				// this can happen if we already entered the interval case when the message is completing.
-				// tab.For(ctx).Info("failed to renew the peek lock", tab.StringAttribute("reason", err.Error()))
+				tab.For(ctx).Error(fmt.Errorf("failed to renew lock: %w", err))
 				return
 			}
-			// tab.For(ctx).Debug("renewed lock success")
-			// listener.Metrics.IncMessageLockRenewedSuccess(message)
+			tab.For(ctx).Debug("renewed lock success")
+			metrics.Processor.IncMessageLockRenewedSuccess(message)
 		case <-ctx.Done():
-			log("Context Done, stop lock renewal")
-			// tab.For(ctx).Info("Stopping periodic renewal")
+			log(ctx, ctx, "context done: stopping periodic renewal")
+			tab.For(ctx).Info("stopping periodic renewal")
 			err := ctx.Err()
 			if errors.Is(err, context.DeadlineExceeded) {
-				// listener.Metrics.IncMessageDeadlineReachedCount(message)
+				metrics.Processor.IncMessageDeadlineReachedCount(message)
 			}
 			alive = false
 		}
 	}
 }
 
-// dumb log until we integrate logging
-func log(a ...any) {
+type Logger interface {
+	Info(s string)
+	Warn(s string)
+	Error(s string)
+}
+
+var getLogger = func(_ context.Context) Logger { return &printLogger{} }
```

Comment:
these are just placeholders right?

Reply:
no, this is the default print logger if you don't set a logger but enable logging.